pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -140,17 +148,27 @@ func (l *LeaderElection) create() (*k8sleaderelection.LeaderElector, error) {
 	if l.remote {
 		key = remoteIstiodPrefix + key
 	}
-	lock := k8sresourcelock.ConfigMapLock{
+	var lock k8sresourcelock.Interface = &k8sresourcelock.ConfigMapLock{
 		ConfigMapMeta: metav1.ObjectMeta{Namespace: l.namespace, Name: l.electionID},
 		Client:        l.client.CoreV1(),
 		LockConfig: k8sresourcelock.ResourceLockConfig{
 			Identity: l.name,
 			Key:      key,
 		},
 	}
+	if l.perRevision {
```

Comment:
Unfortunately we can't easily migrate everything to this because older versions will not work... I wonder if it would be safer to just stick with config map locks for now for consistency.